app/code/community/Razorpay/Payments/Helper/Data.php

```diff
@@ -72,16 +72,18 @@ protected function verifyOrderAmount($razorpayOrderId, $order)
 
     protected function getExpectedRazorpayOrderData($order)
     {
-        $amount  = (int) ($order->getBaseGrandTotal() * 100);
+        $orderCurrency = $order->getBaseCurrencyCode();
 
-        $orderId = $order->getRealOrderId();
+        $orderAmount = (int) ($order->getBaseGrandTotal() * 100);
 
-        $currency = Razorpay_Payments_Model_Paymentmethod::CURRENCY;
+        if ($orderCurrency !== Razorpay_Payments_Model_Paymentmethod::CURRENCY) {
```

Comment:
@mayankamencherla change constant to `BASE_CURRENCY`?

Reply:
@ccrims0n We now support USD as base currency as well. So INR isn't the only base currency.